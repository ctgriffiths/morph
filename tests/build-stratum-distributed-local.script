#!/bin/sh
#
# Test build a simple stratum using as many local workers as makes sense.
#
# Copyright (C) 2012  Codethink Limited
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; version 2 of the License.
# 
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License along
# with this program; if not, write to the Free Software Foundation, Inc.,
# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.

set -e

export PATH="`pwd`:$PATH"

cache="$DATADIR/build-stratum-cache"
log="$DATADIR/build-stratum-morph.log"
./morph --no-default-configs \
    build-distributed morphs-repo master hello-stratum.morph \
    --git-base-url="file://$DATADIR" \
    --cachedir="$cache" --keep-path --no-distcc \
    --log="$log" || cat "$log" 1>&2
tar -tf "$cache/"*.stratum.* | LC_ALL=C sort | sed '/^\.\/./s:^\./::'
