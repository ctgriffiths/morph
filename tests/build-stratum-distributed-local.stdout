./
baserock/
baserock/hello-stratum.meta
baserock/hello.meta
bin/
bin/hello
etc/
